src/main/scala/mesosphere/marathon/state/TaskFailureRepository.scala

```diff
@@ -1,17 +1,20 @@
 package mesosphere.marathon.state
 
-import scala.collection.mutable
+import mesosphere.marathon.metrics.Metrics
 
-class TaskFailureRepository(store: EntityStore[TaskFailure], maxVersions: Option[Int] = Some(1)) {
+import scala.concurrent.Future
 
-  protected[this] val taskFailures = mutable.Map[PathId, TaskFailure]()
+class TaskFailureRepository(
+  protected val store: EntityStore[TaskFailure],
+  protected val maxVersions: Option[Int] = Some(1),
+  protected val metrics: Metrics)
+    extends EntityRepository[TaskFailure] {
 
-  def store(id: PathId, value: TaskFailure): Unit =
-    synchronized { taskFailures(id) = value }
+  def store(id: PathId, value: TaskFailure): Future[TaskFailure] =
+    super.storeWithVersion(id.safePath, value.version, value)
```

Comment:
Why do we store versions of a failure? 
Do we store only the last failure? (Should that be reflected in the class name?)
Should there be one entry per application (and version?) ?